modules/db/src/test/java/org/mule/module/db/internal/domain/connection/SimpleConnectionFactoryTestCase.java

```diff
@@ -44,6 +49,22 @@ public void failsOnConnectionError() throws Exception
 
         connectionFactory.create(dataSource);
     }
+    
+    @Test(expected = ConnectionCreationException.class)
+    public void failsOnConnectionThenLogWithoutCredentials() throws Exception
+    {
+        when(dataSource.getConnection()).thenThrow(new RuntimeException(EXCEPTION_MESSAGE));
+
+        try
+        {
+            connectionFactory.create(dataSource);
+        }
+        catch (ConnectionCreationException e)
+        {
+            assertThat(e.getMessage(), equalTo(EXPECTED_EXCEPTION_MESSAGE));
+            throw e;
```

Comment:
I catch the exception in order to verify the message involved, but an exception is expected and only if ConnectionCreationException is raised the test shall pass (I've changed the mocked behaviour to throw a ConnectionCreationException and not a RuntimeException).